src/levelset/levelSetSegmentationObject.hpp

```diff
@@ -78,6 +78,11 @@ class LevelSetSegmentationSurfaceInfo {
 
     std::array<double, 3> evalNormal(const std::array<double, 3> &point, const SegmentConstIterator &segmentItr) const;
 
+    void evalBarycentricCoordinates(const std::array<double, 3> &point, const SegmentConstIterator &segmentItr, double *lambda) const;
+    void evalProjection(const std::array<double, 3> &point, const SegmentConstIterator &segmentItr, std::array<double, 3> *projectionPoint, std::array<double, 3> *projectionNormal) const;
+    void evalProjection(const std::array<double, 3> &point, const SegmentConstIterator &segmentItr, std::array<double, 3> *projectionPoint) const;
+    std::array<double, 3> evalProjection(const std::array<double, 3> &point, const SegmentConstIterator &segmentItr, double *lambda) const;
```

Comment:
The API is now heterogeneous, can we the evalProjection functions more consistent? For example:
```
std::array<double, 3> evalProjection(const std::array<double, 3> &point, const SegmentConstIterator &segmentItr) const;
std::array<double, 3> evalProjection(const std::array<double, 3> &point, const SegmentConstIterator &segmentItr, std::array<double, 3> *projectionNormal) const;
std::array<double, 3> evalProjection(const std::array<double, 3> &lambda, const SegmentConstIterator &segmentItr) const;
std::array<double, 3> evalProjection(const std::array<double, 3> &lambda, const SegmentConstIterator &segmentItr, std::array<double, 3> *projectionNormal) const;
```

Reply:
The API is heterogeneous because the two "void evalProjection(...)" functions return both the projection point and the normal. I don't like to return the projection point as a return value and the normal as a reference argument. I could change the "std::array<double, 3> evalProjection(...)" to be of type void, but it's a public member and I prefer to keep it as it is.